### DISCUSSION TITLE
Error handling from zValidator validation on frontend using RPC

### DISCUSSION THREAD

**Discussion by callmetwan** [honojs/hono#3430] [General]
When using zValidator and not supplying valid properties to a request, I would expect to get an error response when using RPC. Here is a simple example:

Backend
```typescript
import { Hono } from 'hono'
import { zValidator } from "@hono/zod-validator";
import { z } from "zod";
import { hc } from 'hono/client';

const app = new Hono()

app.get('/', (c) => {
  return c.text('Hello Hono!')
})

const route = app.post(
  '/posts',
  zValidator(
    'form',
    z.object({
      body: z.string().min(1),
    })
  ),
  (c) => {
    const validated = c.req.valid('form')
    return c.json({success: true}, 200)
  }
)

export type AppType = typeof route;
```

Client
```typescript
const client = hc<AppType>('http://localhost:8787/')

const response = await client.posts.$post({
  form: {
    body: '',
  }
})

```

Using zValidator doesn't cause the return type to indicate that a 400 could be returned with errors (which is what [zValidator reutrns](https://github.com/honojs/middleware/blob/main/packages/zod-validator/src/index.ts#L61) ).

Is there something else I need to add to the middleware instantiation? Or is this a known issue?

**Comment by JosephMarotte** [comment]
Hello,
This is related to the following issue : https://github.com/honojs/hono/issues/2719
